### ISSUE TITLE
Left over enigftnexport files in enigma/tmp folder

### ISSUE THREAD

**Issue by Beanzilla** [NuSkooler/enigma-bbs#322]
**Describe the Bug**
There are leftover enigftnexport files in the enigma/tmp folder.

**To Reproduce**
Do `ls -la /tmp`

**Expected Behavior**
Temp files cleaned up after use.

**Actual Behavior**
the tmp directory is full of leftover files.

**Screenshots**
![enigftnissue](https://user-images.githubusercontent.com/20930999/98597907-b36bc780-22a7-11eb-87fc-6980fdde50a6.png)
or
[enigftnissue.txt](https://github.com/NuSkooler/enigma-bbs/files/5513356/enigftnissue.txt)

**Environment**
* [X] I am using Node.js v12.x LTS or higher
* [X] `npm install` or `yarn` reports success
* Actual Node.js version (`node --version`): v12.19.0
* Operating system (`uname -a` on *nix systems): Linux 529761d396dc 5.6.14-x86_64-linode135 #1 SMP PREEMPT Fri May 22 14:57:20 UTC 2020 x86_64 GNU/Linux
* Revision (`git rev-parse --short HEAD`): v0.0.12
* Any additional relevant information about your setup: Running in docker.

**Comment by Beanzilla** [comment]
In enigma/logs I did `grep enigftn enigma-bbs.log` to show all occurrences of enigftnimport and enigftnexport

[enigma-bbs.txt](https://github.com/NuSkooler/enigma-bbs/files/5513381/enigma-bbs.txt)

**Comment by stevet11** [comment]
To debug this, I had to set logging level to trace.
```
[2020-11-09T20:32:27.581-05:00] DEBUG: ENiGMA½ BBS/7 on 529761d396dc: Processing bundle
    bundleFile: {
      "path": "/home/enigma/bbs/mail/ftn_in/8c44843b.mo0",
      "archName": "application/zip"
    }
[2020-11-09T20:32:27.587-05:00]  WARN: ENiGMA½ BBS/7 on 529761d396dc: Failed to extract bundle (path=/home/enigma/bbs/mail/ftn_in/8c44843b.mo0)
    error: External process error: Decompression failed with exit code: 1
[2020-11-09T20:32:27.588-05:00] DEBUG: ENiGMA½ BBS/7 on 529761d396dc: Archiving import file (origPath=/home/enigma/bbs/mail/ftn_in/8c44843b.mo0, type=bundle, status=reject)
    archivePath: /home/enigma/bbs/mail/ftn_reject/reject-bundle--2020-11-09T20.32.27.588-8c44843b.mo0
```
I have unzip and zip, so that isn't the issue.  I tested the zipfile, and it also was ok.

It seemed like it didn't know what to do with the application/zip.

I copied the fileTypes from core/config_default.js into the config.hjson, and we seem to be processing packets fine.  It also looks like we're reusing the same packetPath.

```
[2020-11-09T21:01:12.281-05:00] DEBUG: ENiGMA½ BBS/7 on 529761d396dc: Processing bundle
    bundleFile: {
      "path": "/home/enigma/bbs/mail/ftn_in/8c44843b.mo0",
      "archName": "application/zip"
    }
[2020-11-09T21:01:12.375-05:00]  INFO: ENiGMA½ BBS/7 on 529761d396dc: Import complete (areaFail={}, otherFail=0)
    areaSuccess: {
      "fsx_sta": 1
    }
    --
    packetPath: /tmp/enigftnimport-71bc4a8c7bclmo5nNoc8UuNYzcs/a9f4bd03.pkt
...
[2020-11-09T21:23:56.379-05:00] DEBUG: ENiGMA½ BBS/7 on 529761d396dc: Processing bundle
    bundleFile: {
      "path": "/home/enigma/bbs/mail/ftn_in/8c44843b.mo0",
      "archName": "application/zip"
    }
[2020-11-09T21:23:56.442-05:00]  INFO: ENiGMA½ BBS/7 on 529761d396dc: Import complete (areaFail={}, otherFail=0)
    areaSuccess: {
      "fsx_gen": 1
    }
    --
    packetPath: /tmp/enigftnimport-71bc4a8c7bclmo5nNoc8UuNYzcs/a9fa1119.pkt
...
[2020-11-09T21:40:31.211-05:00] DEBUG: ENiGMA½ BBS/7 on 529761d396dc: Processing bundle
    bundleFile: {
      "path": "/home/enigma/bbs/mail/ftn_in/8c44843b.mo0",
      "archName": "application/zip"
    }
[2020-11-09T21:40:31.263-05:00]  INFO: ENiGMA½ BBS/7 on 529761d396dc: Import complete (areaFail={}, otherFail=0)
    areaSuccess: {
      "fsx_gen": 1
    }
    --
    packetPath: /tmp/enigftnimport-71bc4a8c7bclmo5nNoc8UuNYzcs/a9fdf619.pkt
```

**Comment by stevet11** [comment]
I removed the ftn_secin, and restarted.  More directories created, but unrelated.

This is running in a docker container.  Every container restart, two directories /tmp/enigftn(export|import)-* are created.  After many restarts, we have many directories.

I've changed the way docker handles the /tmp directory.

**Comment by NuSkooler** [comment]
The /tmp/ directories should be created and cleaned up on startup/shutdown (e.g.: https://github.com/NuSkooler/enigma-bbs/blob/master/core/scanner_tossers/ftn_bso.js#L2184). They are used as a temporary extraction location for FTN packets.

It sounds like your Docker container is not cleanly shutting down the system -- thus not going through the shutdown path. Might double check that.

As for the `importDir` issue, that's a separate issue. It's a debug message that I plan on cleaning up at some point, but it's harmless.

**Comment by stevet11** [comment]
Here's what I know so far:

inside of docker using interactive shell (avoiding all of the PID1/signal handling strangeness of docker), sending ctrl-c, using kill to send SIGINT or SIGTERM results in enigma stopping.  
As in nothing is displayed after the "System started!" line.   

```
node@129977c79207:/home/enigma/bbs$ node main.js 
ENiGMA½ Copyright (c) 2015-2020, Bryan D. Ashby
_____________________   _____  ____________________    __________\_   /
\__   ____/\_ ____   \ /____/ /   _____ __         \  /   ______/ // /___jp!
//   __|___//   |    \//   |//   |    \//  |  |    \//        \ /___   /_____
/____       _____|      __________       ___|__|      ____|     \   /  _____  \
---- \______\ -- |______\ ------ /______/ ---- |______\ - |______\ /__/ // ___/
                                                                     /__   _\
 <*>   ENiGMA½  // HTTPS://GITHUB.COM/NUSKOOLER/ENIGMA-BBS   <*>       /__/

-------------------------------------------------------------------------------

System started!
^C
node@129977c79207:/home/enigma/bbs$
```
However, this "works" with the above session:
```
node@129977c79207:/home/enigma/bbs$ node oputil.js config new
? Create a new configuration? (y/N) y
? Create a new configuration? Yes
? Configuration path: (./config/config.hjson) 
? Configuration path: ./config/config.hjson
? Config file exists. Overwrite? (y/N) y
? Config file exists. Overwrite? Yes
? BBS name: (New ENiGMA½ BBS) 
? BBS name: New ENiGMA½ BBS
? First message conference: (Local) 
? First message conference: Local
? Conference description: (Local Areas) 
? Conference description: Local Areas
? First area in message conference: (General) 
? First area in message conference: General
? Area description: (General chit-chat) 
? Area description: General chit-chat
? Logging level: 
? Logging level: Trace
Configuration generated

node@129977c79207:/home/enigma/bbs$ node main.js 
ENiGMA½ Copyright (c) 2015-2020, Bryan D. Ashby
_____________________   _____  ____________________    __________\_   /
\__   ____/\_ ____   \ /____/ /   _____ __         \  /   ______/ // /___jp!
//   __|___//   |    \//   |//   |    \//  |  |    \//        \ /___   /_____
/____       _____|      __________       ___|__|      ____|     \   /  _____  \
---- \______\ -- |______\ ------ /______/ ---- |______\ - |______\ /__/ // ___/
                                                                     /__   _\
 <*>   ENiGMA½  // HTTPS://GITHUB.COM/NUSKOOLER/ENIGMA-BBS   <*>       /__/

-------------------------------------------------------------------------------

System started!
^CProcess interrupted. Shutting down...
Goodbye!
node@129977c79207:/home/enigma/bbs$ 
```
Using our "live" config.hjson, I've tried turning off everything (but telnet).  I've reset back to blocktronics theme.  Still results in improper shutdown.

**Ok!**  It isn't _docker_.  I just added enigma to my desktop here so I could keep my config file paths the same.  It's something in my setup/config files.

**Comment by stevet11** [comment]
Ok, deleting the events section fixes it.  Or, having events that don't run.  
Once an event runs, it breaks.

Or, if I disable events, but run a door, it breaks.
"it breaks" = stopping enigma I don't get the proper "Process interrupted. Shutting down..." message.

**Comment by stevet11** [comment]
See https://github.com/microsoft/node-pty/issues/190 for the exact problem.

Fixed in https://github.com/microsoft/node-pty/pull/218

**Comment by NuSkooler** [comment]
Will need to wait on https://github.com/microsoft/node-pty/pull/218 to be put into a release. We're currently at the latest version.

Until then you can force your local version to use e.g. the fix branch or node-pty master by modifying your `package.json`

**Comment by NuSkooler** [comment]
The Microsoft/node-pty team has responded regarding this - using the beta tag should work around the issue for now.

> @NuSkooler you can use the beta tag to get the latest version, no date for 0.10 yet

You can do this via npm or yarn install or manual edit to your package.json for now.

**Comment by NuSkooler** [comment]
Microsoft has updated node-pty to v0.10.0 and ENiGMA has followed. Should now be resolved. Feel free to re-open if this is still a problem after updating!
